terraform/java/ec2/default/main.tf

```diff
@@ -138,6 +138,19 @@ resource "null_resource" "main_service_setup" {
       # The application needs time to come up and reach a steady state, this should not take longer than 30 seconds
       sleep 30
 
+      # Check if the application is up. If it is not up, then exit 1.
+      attempt_counter=0
+      max_attempts=30
+      until $(curl --output /dev/null --silent --head --fail $(echo "http://localhost:8080" | tr -d '"')); do
+        if [ $attempt_counter -eq $max_attempts ];then
+          echo "Failed to connect to endpoint. "
+          exit 1
+        fi
+        printf 'Attempting to connect to the endpoint. Tried $attempt_counter out of $max_attempts'
```

Comment:
use echo, `printf` will append and not start a new line I think

Reply:
And specify which endpoint (main vs remote)

Reply:
Done